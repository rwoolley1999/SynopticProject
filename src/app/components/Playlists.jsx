// import React, { Component } from "react";
// import { MDBBtn, MDBInput, MDBModal, MDBModalBody, MDBModalHeader, MDBModalFooter, MDBIcon, MDBBadge, MDBContainer, MDBRow, MDBCol} from "mdbreact";

// export class Playlists extends Component {
//   constructor(props) {
//     super(props);
//     this.state = {
//       modal: false,
//       playlists: [
//         {
//           id: 1,
//           name: "Example Playlist",
//           category: "Pop",
//           comment: "This is an example playlist"
//         }
//       ]
//     };
//   }
//   addPlaylist = () => {
//     var newArray = [...this.state.playlists];
//     newArray.push({
//       id: newArray.length ? newArray[newArray.length - 1].id + 1 : 1,
//       name: this.state.name,
//       category: this.state.category,
//       comment: this.state.comment
//     });
//     this.setState({ playlists: newArray });
//     this.setState({
//       name: "",
//       category: "",
//       comment: ""
//     });
//   };

//   handleInputChange = inputName => value => {
//     const nextValue = value;

//     this.setState({
//       [inputName]: nextValue
//     });
//   };

//   handleDelete = playlistId => {
//     const playlists = this.state.playlists.filter(e => e.id !== playlistId);
//     this.setState({ playlists });
//   };
//   toggleModal = () => {
//     this.setState({
//       modal: !this.state.modal
//     });
//   };

//   render() {
//     return (
//       <React.Fragment>
//         <MDBContainer>
//           <MDBRow>
//             <MDBCol md="9" className="mb-r">
//               <div id="schedule-items">
//                 {this.state.playlists.map(playlist => (
//                   <PlaylistManagement
//                     key={playlist.id}
//                     id={playlist.id}
//                     name={playlist.name}
//                     category={playlist.category}
//                     comment={playlist.comment}
//                     onDelete={this.handleDelete}
//                   />
//                 ))}
//               </div>
// 			   <MDBRow className="mb-4">
//                 <MDBCol xl="3" md="6" className="mx-auto text-center">
//                   <MDBBtn color="info" rounded onClick={this.toggleModal}>
//                     Add Event
//                   </MDBBtn>
//                 </MDBCol>
//               </MDBRow>
//               </MDBCol>
//               </MDBRow>
//         </MDBContainer>

//         <MDBModal isOpen={this.state.modal} toggle={this.toggleModal}>
//           <MDBModalHeader
//             className="text-center"
//             titleClass="w-100 font-weight-bold"
//             toggle={this.toggleModal}
//           >
//             Add new playlist
//           </MDBModalHeader>
//           <MDBModalBody>
//             <form className="mx-3 grey-text">
//               <MDBInput
//                 name="name"
//                 label="Playlist Name"
//                 hint="Bach Collection"
//                 group
//                 type="text"
//                 getValue={this.handleInputChange("name")}
//               />
//               <MDBInput
//                 name="category"
//                 label="Category (optional)"
//                 hint="Classical"
//                 group
//                 type="text"
//                 getValue={this.handleInputChange("category")}
//               />
//               <MDBInput
//                 name="comment"
//                 label="Comment (optional)"
//                 group
//                 type="textarea"
//                 getValue={this.handleInputChange("comment")}
//               />
//             </form>
//           </MDBModalBody>
//           <MDBModalFooter className="justify-content-center">
//             <MDBBtn
//               color="info"
//               onClick={() => {
//                 this.toggleModal();
//                 this.addPlaylist();
//               }}
//             >
//               Add
//             </MDBBtn>
//           </MDBModalFooter>
//         </MDBModal>
//       </React.Fragment>
//     );
//   }
// }

// class PlaylistManagement extends Component {
//   render() {
//     return (
//       <React.Fragment>
//         <div className="media mt-1">
//           <h3 className="h3-responsive font-weight-bold mr-3">
//             {this.props.name}
//           </h3>
//           <div className="media-body mb-3 mb-lg-3">
//             <MDBBadge
//               color="danger"
//               className="ml-2 float-right"
//               onClick={() => this.props.onDelete(this.props.id)}
//             >
//               -
//             </MDBBadge>
//             <h6 className="mt-0 font-weight-bold">{this.props.category} </h6>
//             <hr className="hr-bold my-2" />
//           </div>
//         </div>
//         {this.props.comment && (
//           <p className="p-2 mb-4  blue-grey lighten-5 blue-grey lighten-5">
//             {this.props.comment}
//           </p>
//         )}
//       </React.Fragment>
//     );
//   }
// }

import React from 'react';
import { connect } from 'react-redux';
import  { requestPlaylistCreation } from '../store/mutations';
import { Link } from 'react-router-dom';

export const Playlists = ({playlists, name, id, createNewPlaylist})=>(
    <div>
        <h3>
            {name}
        </h3>
        {playlists.map(playlist=>(
            <Link to={'/playlist/'+playlist.id} key={playlist.id}>
              <div>{playlist.name}</div>
            </Link>
              ))}
        <button onClick={()=>createNewPlaylist(id)}>Add New Playlist</button>
    </div>
);

const mapStateToProps=(state, ownProps)=>{
    let categoryID = ownProps.id;
    return{
        name:ownProps.name,
        id:categoryID,
        playlists:state.playlists.filter(playlist=>playlist.category === categoryID)
    }
}

const mapDistpatchToProps = (dispatch, ownProps)=>{
    return{
        createNewPlaylist(id){
            console.log("Creating new task", id);
            dispatch(requestPlaylistCreation(id));
        }
    }
}
export const ConnectedPlaylists = connect(mapStateToProps, mapDistpatchToProps)(Playlists);
