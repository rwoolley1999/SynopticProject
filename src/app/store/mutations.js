export const REQUEST_PLAYLIST_CREATION = 'REQUEST_PLAYLIST_CREATION';
export const CREATE_PLAYLIST = 'CREATE_PLAYLIST';
export const ASSIGN_PLAYLIST_NAME = "ASSIGN_PLAYLIST_NAME"
export const DELETE_PLAYLIST = "DELETE_PLAYLIST";
export const REQUEST_NEW_CATEGORY = "REQUEST_NEW_CATEGORY";
export const ADD_NEW_CATEGORY = "ADD_NEW_CATEGORY";
export const ASSIGN_CATEGORY_NAME = "ASSIGN_CATEGORY_NAME";
export const DELETE_CATEGORY = "DELETE_CATEGORY";
export const REQUEST_ADD_FILE = "REQUEST_ADD_FILE";
export const ADD_MEDIA_FILE = "ADD_MEDIA_FILE";
export const ASSIGN_MEDIA_FILE_NAME = "ASSIGN_MEDIA_FILE_NAME";
export const ASSIGN_MEDIA_FILE_CATEGORY = "ASSIGN_MEDIA_FILE_CATEGORY";
export const ASSIGN_MEDIA_FILE_PLAYLIST = "ASSIGN_MEDIA_FILE_PLAYLIST";
export const ADD_MEDIA_FILE_COMMENT = "ADD_MEDIA_FILE_COMMENT";
export const ASSIGN_MEDIA_FILEPATH = "ASSIGN_MEDIA_FILEPATH";
export const ADD_MEDIA_FILE_IMAGE = "ADD_MEDIA_FILE_IMAGE";
export const DELETE_MEDIA_FILE = "DELETE_MEDIA_FILE";

export const requestPlaylistCreation = () => ({
    type: REQUEST_PLAYLIST_CREATION
});

export const createPlaylist = (playlistID) => ({
    type: CREATE_PLAYLIST,
    playlistID
});

export const assignPlaylistName = (id, name) => ({
    type: ASSIGN_PLAYLIST_NAME,
    playlistID: id,
    name
});

export const deletePlaylist = (playlistID) => ({
    type: DELETE_PLAYLIST,
    playlistID
});

export const requestCategoryCreation = () => ({
    type: REQUEST_NEW_CATEGORY
});

export const addNewCategory = (categoryID) => ({
    type: ADD_NEW_CATEGORY,
    categoryID
});

export const assignCategoryName = (id, name) => ({
    type: ASSIGN_CATEGORY_NAME,
    categoryID: id,
    name
});

export const deleteCategory = (categoryID) => ({
    type: DELETE_CATEGORY,
    categoryID
});

export const requestFileAddition = (playlistID) => ({
    type: REQUEST_ADD_FILE,
    playlistID
});

export const addMediaFile = (mediaFileID, playlistID) => ({
    type: ADD_MEDIA_FILE,
    mediaFileID,
    playlistID
});

export const assignMediaFileName = (id, name) => ({
    type: ASSIGN_MEDIA_FILE_NAME,
    mediaFileID: id,
    name
});

export const assignMediaFileCategory = (id, categoryID) => ({
    type: ASSIGN_MEDIA_FILE_CATEGORY,
    mediaFileID: id,
    categoryID
});

export const assignMediaFilePlaylist = (id, playlistID) => ({
    type: ASSIGN_MEDIA_FILE_PLAYLIST,
    mediaFileID: id,
    playlistID
});

export const addMediaFileComment = (id, comment) => ({
    type: ADD_MEDIA_FILE_COMMENT,
    mediaFileID: id,
    comment
});

export const assignMediaFilePath = (id, filePath) => ({
    type: ASSIGN_MEDIA_FILEPATH,
    mediaFileID: id,
    filePath
});

export const addMediaFileImage = (id, imageURL) => ({
    type: ADD_MEDIA_FILE_IMAGE,
    mediaFileID: id,
    imageURL
});

export const deleteMediaFile = (mediaFileID) => ({
    type: DELETE_MEDIA_FILE,
    mediaFileID
});