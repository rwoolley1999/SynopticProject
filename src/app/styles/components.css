body {
    background-color: #282c34;
    color: white;
    padding: 40px;
    font-family: Arial;
    text-align: center;
  }

  .button {
    background-color: white; 
    border-radius: 4px;
    color: #282c34;
    padding: 5px 10px;
    text-align: center;
    display: inline-block;
    font-size: 16px;
    transition-duration: 0.4s;
    margin: 8px 0;
  }

  .button:hover {
    background-color: #4169E1; /* Green */
    color: white;
  }

  input {
    width: 20%;
    padding: 9px 10px;
    margin: 8px 0;
    box-sizing: border-box;
  }

  /* .column{
    column-count: 2;
    column-gap: 0px;
  } */

  /* FINISH ADDING CSS */