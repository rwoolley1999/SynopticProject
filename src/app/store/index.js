import { createStore, applyMiddleware, combineReducers } from 'redux';
import { defaultState } from './default_state';
import { createLogger } from 'redux-logger';
import * as mutations from './mutations';
import createSagaMiddleware from 'redux-saga';
import * as sagas from './sagas';
import {loadState, saveState } from '../state/local_storage';
// import { saveFile } from '../state/saveFile';

const sagaMiddleware = createSagaMiddleware();

export const persistedState = loadState();

export const store = createStore(
    combineReducers({
        playlists(playlists = defaultState.playlists, action){
            switch(action.type){
                case mutations.CREATE_PLAYLIST:
                    return[...playlists, {
                            id: action.playlistID,
                            name: "New Playlist"
                    }]
                case mutations.ASSIGN_PLAYLIST_NAME:
                    return playlists.map(playlist=>{
                        return (playlist.id === action.playlistID) ?
                        {...playlist, name: action.name} :
                        playlist;
                    });
                case mutations.DELETE_PLAYLIST:
                    return playlists.filter(playlist=>{
                        return(playlist.id !== action.playlistID);
                    });
            }
            return playlists;
        },
        categories(categories = defaultState.categories, action){
            switch(action.type){
            case mutations.ADD_NEW_CATEGORY:
                return[...categories, {
                        id: action.categoryID,
                        name: "New Category"
                }]
            case mutations.ASSIGN_CATEGORY_NAME:
                return categories.map(category=>{
                    return (category.id === action.categoryID) ?
                        {...category, name: action.name} :
                        category;
                    });
            case mutations.DELETE_CATEGORY:
                return categories.filter(category=>{
                    return(category.id !== action.categoryID);
                    });
        }
        return categories;
        },
        mediaFiles(mediaFiles = defaultState.mediaFiles, action){
            switch(action.type){
                case mutations.ADD_MEDIA_FILE:
                    return[...mediaFiles, {
                            id: action.mediaFileID,
                            name: "New Media File",
                            playlist:action.playlistID
                    }]
                case mutations.ASSIGN_MEDIA_FILE_NAME:
                    return mediaFiles.map(mediaFile=>{
                        return (mediaFile.id === action.mediaFileID) ?
                            {...mediaFile, name: action.name} :
                            mediaFile;
                    });
                case mutations.ASSIGN_MEDIA_FILE_CATEGORY:
                    return mediaFiles.map(mediaFile=>{
                        return (mediaFile.id === action.mediaFileID) ? 
                            {...mediaFile, category:action.categoryID} : 
                            mediaFile;
                    });
                case mutations.ASSIGN_MEDIA_FILE_PLAYLIST:
                    return mediaFiles.map(mediaFile=>{
                        return (mediaFile.id === action.mediaFileID) ? 
                            {...mediaFile, playlist:action.playlistID} : 
                            mediaFile;
                    });
                case mutations.ADD_MEDIA_FILE_COMMENT:
                    return mediaFiles.map(mediaFile=>{
                        return (mediaFile.id === action.mediaFileID) ? 
                            {...mediaFile, comment:action.comment} : 
                            mediaFile;
                    });
                case mutations.ASSIGN_MEDIA_FILEPATH:
                    return mediaFiles.map(mediaFile=>{
                        return (mediaFile.id === action.mediaFileID) ?
                            {...mediaFile, filePath: action.filePath} :
                            mediaFile;
                    });
                case mutations.DELETE_MEDIA_FILE:
                    return mediaFiles.filter(mediaFile=>{
                        return(mediaFile.id !== action.mediaFileID);
                    });
        }
            return mediaFiles;
    },
    }),
    persistedState,
    applyMiddleware(createLogger(), sagaMiddleware)
);

store.subscribe(()=>{
    saveState(store.getState());
});


for (let saga in sagas){
    sagaMiddleware.run(sagas[saga]);
}
