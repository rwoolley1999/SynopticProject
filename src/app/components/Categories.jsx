import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import  { requestCategoryCreation } from '../store/mutations';

export const Categories = ({ name, id, categories, addNewCategory })=>(
    <div>
        {categories.map(category=>(
            <Link to={'/categories/'+category.id} key={category.id}>
            <div>
                {category.name}
            </div>
          </Link>
        ))}
         <button onClick={()=>addNewCategory(id)} >Add New Category</button> 
    </div>
);

const mapStateToProps=(state, ownProps)=>{
    let id = ownProps.id;
    return{
        name:ownProps.name,
        id:id,
        categories:state.categories
    }
}

const mapDistpatchToProps = (dispatch, ownProps)=>{
    return{
        addNewCategory(id){
            console.log("Creating new category", id);
            dispatch(requestCategoryCreation(id));
        }
    }
}

export const ConnectedCategories = connect(mapStateToProps, mapDistpatchToProps)(Categories);
