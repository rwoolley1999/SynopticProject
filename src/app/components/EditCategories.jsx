//CONVERT THIS INTO A MODAL IF HAVE TIME
import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import * as mutations from '../store/mutations';

const EditCategories = ({
    id,
    category,
    categories,

    assignCategoryName,
    deleteCategory
})=>{
    return(
    <div>
        <div>
            <input onChange={assignCategoryName} value={category.name}/>
        </div>
        <div>
        <Link to="/">
            <button onClick={()=>deleteCategory(id)} value={category.id}>Delete Category</button>
        </Link>
        </div>
        <div>
            <Link to="/">
                <button>Done</button>
            </Link>
        </div>
    </div>
    )};
    


const mapStateToProps=(state, ownProps)=>{
    let id = ownProps.match.params.id;
    let category = state.categories.find(category=>category.id === id);

    return{
        id,
        category
    }
};

const mapDispatchToProps = (dispatch, ownProps)=>{
    const id = ownProps.match.params.id;
    return{
        assignCategoryName(e){
            dispatch(mutations.assignCategoryName(id, e.target.value));
        },
        deleteCategory(id){
            console.log("deleting category", id);
            dispatch(mutations.deleteCategory(id));
        }
    }
}

export const ConnectedEditCategories = connect(mapStateToProps, mapDispatchToProps)(EditCategories);