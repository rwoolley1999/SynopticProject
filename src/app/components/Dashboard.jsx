import React from 'react';
import { connect } from 'react-redux';
import { ConnectedPlaylists } from './playlists';
import { Link } from 'react-router-dom';
import '../styles/components.css';

export const Dashboard = ({playlists})=>(
    <div>
    <div class="column">
        <Link to={'/categories'}>
        <div>
                Categories
            </div>
        </Link>
        <Link to={'/mediafiles'}>
            <div>
                Media Files
            </div>
        </Link>
    </div>
        <h2>Playlists</h2>
        <ConnectedPlaylists/>
    </div>
);

function mapStateToProps(state){
    return{
        playlists:state.playlists
    }
}

export const ConnectedDashboard = connect(mapStateToProps)(Dashboard);

//got up to start of routing amd navigation vid. Start adding in this. See fav for save/load state

//FIX SO PLAYLISTS ARE GROUPED BY CATEGORY