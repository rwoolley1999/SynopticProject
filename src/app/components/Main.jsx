// import React, {Component} from 'react';
// import { BrowserRouter as Router, Route } from 'react-router-dom';
// import { Dashboard } from './Dashboard';
// import { Playlists } from './Playlists';
// import { Navigation } from './Navigation';

// export class Main extends Component{
//     render(){
//         return(
//         <Router>
//         <div>
//             <Navigation/>
//             <Route exact path='/' component={Dashboard} />
//             <Route exact path='/playlist/:id' component={Playlists}/>
//             {/* fix this so it links to list of media files added to playlist and so that it acc gets the playlist id */}
//         </div>
//         </Router>
//         )
//     }
// }

import React from 'react';
import {Router, Route} from 'react-router-dom';
import { Provider } from 'react-redux';
import { store } from '../store';
import {ConnectedDashboard } from './Dashboard';
import { history } from '../store/history';
import { ConnectedNavigation } from './Navigation';
import { ConnectedPlaylistItems } from './PlaylistItems';
import { ConnectedCategories } from './Categories';
import { ConnectedEditCategories } from './EditCategories';
import { ConnectedMedia } from './MediaFiles';
import { ConnectedEditMedia } from './EditMedia';

export const Main  = () =>(
    <Router history={history}>
        <Provider store={store}>
            <div>
                <ConnectedNavigation/>
                <Route exact path="/" component={ConnectedDashboard} />
                <Route exact path="/playlist/:id" component={({match})=>(<ConnectedPlaylistItems match={match}/>)}/>
                <Route exact path="/categories" component={ConnectedCategories}/>
                <Route exact path="/categories/:id" component={({match})=>(<ConnectedEditCategories match={match}/>)}/>
                <Route exact path="/mediafiles" component={ConnectedMedia}/>
                <Route exact path="/mediafiles/:id" component={({match})=>(<ConnectedEditMedia match={match}/>)}/>
            </div>
        </Provider>
    </Router>
)