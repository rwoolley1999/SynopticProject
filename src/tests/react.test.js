import React from 'react';
import ReactDOM from 'react-dom';
import { Main } from '../app/components/Main';
import { ConnectedPlaylists } from '../app/components/Playlists';
import { shallow, mount } from 'enzyme';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import { Link,MemoryRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import enzymeConfig from './enzymeConfig';
import { deleteCategory } from '../app/store/mutations';
import { ConnectedEditCategories} from '../app/components/EditCategories';

it("renders", () => {
    shallow(<Main />);
  });

//   it("displays initial to-dos", () => {
//     const wrapper = mount(<Playlists />);
//     expect(wrapper.find("h3")).toHaveLength(1);
//   });

  const mockStore = configureStore([]);
  describe('My Connected React-Redux Component', () => {
    let store;
    let component;
    beforeEach(() => {
      store = mockStore({
        myState: 'sample text',
      });
      store.dispatch = jest.fn();

      component = renderer.create(
        <Provider store={store}>
          <ConnectedEditCategories />
        </Provider>
      );
    });

    it('should render with given state from Redux store', () => {
        expect(component.toJSON()).toMatchSnapshot();
    });

    
    it('should insert value to input field', () => {
    renderer.act(() => {
        component.root.findByType('input')
          .props.onChange({ target: { value: 'some other text' } });
      });
    });

    it('should dispatch an action on button click', () => {
        renderer.act(() => {
            component.root.findByType('button').props.onClick();
          });
          expect(store.dispatch).toHaveBeenCalledTimes(1);
    });

    it('Link Exists', () => {

        expect(component.root.findByType(Link)).toBeTruthy();
        // expect(component.root.findByType(Link).toEqual('/'));
      });
  });