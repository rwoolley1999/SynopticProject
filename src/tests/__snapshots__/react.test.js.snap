// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`My Connected React-Redux Component should render with given state from Redux store 1`] = `
<div>
  <div>
    <input
      onChange={[Function]}
    />
  </div>
  <div>
    <a
      href="/"
      onClick={[Function]}
    >
      <button
        id="delete"
        onClick={[Function]}
      >
        Delete Category
      </button>
    </a>
  </div>
</div>
`;
